src/App.js

```diff
@@ -1,70 +1,20 @@
 import React, { useState, useEffect } from "react";
-import API from "./api";
-import Card from "./components/Card";
+import getProduct from "./api/products";
+import ProductsView from "./views/ProductsView";
 import './App.css';
 
 function App() {
   const [products, setProducts] = useState([]);
-  const [searchInput, setSearchInput] = useState("");
-  const [searchProduct, setProductInput] = useState("");
 
   useEffect(() => {
-    API.getProduct().then((res) => {
+    getProduct().then((res) => {
       setProducts(res.data);
     });
-  });
-
-  const cardResults = [];
-
-  for (let i = 0; i < products.length; i++) {
-    if (searchProduct || searchInput) {
-      if (searchProduct && products[i].type.match(searchProduct)) {
-        cardResults.push(<Card cardResults={products[i]} />);
-      }
-      if (searchInput && products[i].name.match(searchInput)) {
-        cardResults.push(<Card cardResults={products[i]} />);
-      }
-    } else {
-      cardResults.push(<Card cardResults={products[i]} />);
-    }
-  }
+  }, []);
```

Comment:
Here i added an empty array so that the api call only happens when the application loads. Without this argument, the useEffect hook will just keep running itself over and over again on every re-render. We should use this argument area to have the hook watch for changes in specific states or areas in our app we want it to trigger from.